File: packages/control-plane/test/fixtures/index/sample.php
<?php

// fixture：PHP（interface / class / 构造器 / 方法 / 常量；属性刻意不收）。

const MAX_RETRIES = 3;

interface Reader
{
    public function read(string $key): string;
}

class Store implements Reader
{
    private string $name;

    public function __construct(string $name)
    {
        $this->name = $name;
    }

    public function read(string $key): string
    {
        return $this->name . $key;
    }
}

function open(string $name): Store
{
    return new Store($name);
}

File: packages/control-plane/test/fixtures/index/sample.rs
//! fixture：Rust（struct / enum / trait / impl 里的方法 / 自由函数）。

pub const MAX_RETRIES: u32 = 3;

pub struct Store {
    name: String,
}

pub enum Mode {
    Fast,
    Slow,
}

pub trait Reader {
    fn read(&self, key: &str) -> String;
}

impl Store {
    pub fn new(name: &str) -> Self {
        Self {
            name: name.to_string(),
        }
    }

    pub fn read(&self, key: &str) -> String {
        format!("{}{}", self.name, key)
    }
}

pub fn open(name: &str) -> Store {
    Store::new(name)
}
